sway/config/output.c

```diff
@@ -164,23 +164,31 @@ static void set_mode(struct wlr_output *output, int width, int height,
 	}
 }
 
-void terminate_swaybg(pid_t pid) {
-	int ret = kill(pid, SIGTERM);
-	if (ret != 0) {
-		wlr_log(WLR_ERROR, "Unable to terminate swaybg [pid: %d]", pid);
-	} else {
-		int status;
-		waitpid(pid, &status, 0);
+static void terminate_swaybg(struct wl_display *display, pid_t pid) {
+	struct wl_client *client;
+	wl_client_for_each(client, wl_display_get_client_list(display)) {
+		pid_t p;
+		wl_client_get_credentials(client, &p, NULL, NULL);
```

Comment:
`wl_client_get_credentials` can fail (e.g. on FreeBSD where it's not supported). We should either check the return value or initialize `p` to 0 to make sure we don't kill random clients.

Reply:
Looking at the freebsd patches for libwayland, it will simply set the pid to 0, which won't break the logic of this loop; it just won't close the connection.

wl_client_get_credentials has no return value. It's just a getter for values which are read right when the client connects.

The only other way I can think to track this is through the socket file descriptor. It would require us opening it ourselves and using WAYLAND_SOCKET, and would require a fair bit of reworking. Not to mention it doesn't work if a process wants multiple connections (do we want to allow that?).
Checking for the pid seems like to most reliable way, at least for Linux.

Reply:
>wl_client_get_credentials has no return value

Oh, all right then, I somehow had the feeling it could fail but I've mixed things up.